xarray/core/dataset.py

```diff
@@ -1328,6 +1330,12 @@ def _validate_indexers(self, indexers):
                 raise TypeError('cannot use a Dataset as an indexer')
             else:
                 v = np.asarray(v)
+
+                if (v.dtype.kind == 'U'
```

Comment:
only the first half is really required (the tests all pass), but I added the second bit to be safe.